### PR TITLE
[BugFix] Fix Iceberg RestCatalog FileIO NPE problem

### LINKED ISSUE #23432: Using Iceberg external catalog.  Gives issues when trying to query data. Unable to load region from system settings.
Following instructions at https://github.com/StarRocks/starrocks/discussions/23427

Following the Apache Iceberg quickstart at https://iceberg.apache.org/spark-quickstart/#adding-a-catalog and modifying the docker compose yaml to include the SR allin1 image.   Did the rest of the iceberg quickstart (creating table and inserting data).     Created the external catalog inside of SR but getting region error when I try to query the iceberg tables.

```
StarRocks > create external catalog 'iceberg'
    -> PROPERTIES
    -> (
    ->     "aws.s3.use_instance_profile" = "true",
    ->     "aws.s3.region" = "us-east-1",
    ->     "uri" = "http://iceberg-rest:8181",
    ->     "warehouse" = "starrocks",
    ->     "type" = "iceberg",
    ->     "iceberg.catalog.type" = "rest"
    -> );
Query OK, 0 rows affected (0.02 sec)

StarRocks > set catalog iceberg;
Query OK, 0 rows affected (0.01 sec)

StarRocks > use nyc
Database changed
StarRocks > select * from taxis;
ERROR 1064 (HY000): Unable to load region from any of the providers in the chain software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain@334f960c: [software.amazon.awssdk.regions.providers.SystemSettingsRegionProvider@5dd640b6: Unable to load region from system settings. Region must be specified either via environment variable (AWS_REGION) or  system property (aws.region)., software.amazon.awssdk.regions.providers.AwsProfileRegionProvider@5f0eb1d8: No region provided in profile: default, software.amazon.awssdk.
StarRocks > show tables;
+---------------+
| Tables_in_nyc |
+---------------+
| taxis         |
+---------------+
1 row in set (0.03 sec)

```

### PR DESCRIPTION
**Pull request by Smith-Cruise**
Fixes #23432

Now we can use Tabular via Iceberg RestCatalog smoothly.

```sql
create external catalog tabular properties (
	"type"="iceberg",
	"iceberg.catalog.type"="rest",
	"uri"="https://api.tabular.io/ws",
	"credential"="t-5Ii8T9m0",
	"warehouse"="smith",
	"aws.s3.region"="us-west-2",
	"aws.s3.access_key"="AKIAMST",
	"aws.s3.secret_key"="LJUchAtmacTuf+"
);
```

```sql
set catalog tabular;
```

```sql
mysql> show databases;
+----------+
| Database |
+----------+
| default  |
| examples |
| hello    |
| sr       |
| system   |
+----------+
```

```sql
use sr;

create table hello_smith (id int, name string);

mysql> insert into hello_smith values (1, "smith"), (2, "cruise");
Query OK, 2 rows affected (8.99 sec)
{'label':'FAKE_ICEBERG_SINK_LABEL', 'status':'VISIBLE', 'txnId':'-1'}

mysql> select * from hello_smith;
+------+--------+
| id   | name   |
+------+--------+
|    1 | smith  |
|    2 | cruise |
+------+--------+
2 rows in set (1.66 sec)
```

### CODE REVIEW

**Review by imay — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java|0|1|00.00%|[127]|

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java [hunk 1]
```diff
@@ -123,6 +123,8 @@ public void initialize(Map<String, String> properties) {
                     break;
                 case "glue":
                     wrappedIO = new S3FileIO();
```
Comment (zombee0) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java:125]: break?

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java]: fixed

**Review by zombee0 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**